src/store/modules/maargJob/actions.ts

```diff
@@ -0,0 +1,122 @@
+import { ActionTree } from 'vuex'
+import RootState from '@/store/RootState'
+import JobState from './MaargJobState'
+import * as types from './mutation-types'
+import { isCustomRunTime, generateAllowedFrequencies, hasError, showToast } from '@/utils'
+import store from '@/store'
+import logger from "@/logger";
+import { MaargJobService } from '@/services/MaargJobService'
+
+const actions: ActionTree<JobState, RootState> = {
+  async fetchMaargJobs({ commit, dispatch }, jobTypeEnumIds){
+    const productStoreId = store.getters["user/getCurrentEComStore"]?.productStoreId
+    const maargJobEnums = store.getters["maargJob/getMaargJobEnums"]
+
+    let resp = {} as any;
+    const maargJobs = {} as any;
+
+    try {
+      resp = await MaargJobService.fetchMaargJobs({ jobTypeEnumId: jobTypeEnumIds, jobTypeEnumId_op: "in" })
+      if(!hasError(resp) && resp.data?.length) {
+        const jobs = resp.data;
+        const jobEnumIdsToFetch = jobs.map((job: any) => job.jobTypeEnumId);
+        
+        const responses = await Promise.allSettled(jobs.map((job: any) => MaargJobService.fetchMaargJobInfo(job.jobName)))
+        await dispatch("fetchMaargJobEnums", jobEnumIdsToFetch)
```

Comment:
I think we should move this outside of Promise, as it will be called every time the loop will run, but its independent of the loop.